packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -122,6 +122,157 @@ The container for the right-aligned actions like search, buttons, and user profi
 
 ### TabNav
 
+The horizontal navigation bar that contains multiple navigation items, each represented by TabNavItem. 
+
+TabNav's interaction is as follows: 
+
+1. There might already be items under More even at the highest screen size. These items might need a custom order.
+2. Once the screen size starts getting reduced, more items will start moving into More.
+3. The order of these items would follow a logical order. They keep getting added ahead of the already existing items. So the last item to go into More is the first item that you see when you click on More.
+
+TabNav automatically handles all these edge cases & responsiveness.
+
+```jsx
+<TabNav
+  items={[
```

Comment:
> why do we need to both pass the items here and then also specific which component to use to render this data? 🤔

[Why was this prop-based API chosen over the old compound component API?
](https://github.com/razorpay/blade/blob/anu/tabnav-newapi/packages/blade/src/components/TopNav/_decisions/decisions.md#why-was-this-prop-based-api-chosen-over-the-old-compound-component-api)


The TabNav's interaction requires us to control what items are shown horizontally and what items goes inside the "more" menu dynamically depending on the screen width. 

And since we want to keep the flexibility of letting consumers choose what/how things gets rendered inside the More menu we need to accept the data item object instead of using compound API. 

For example: 
An item say "Payroll" doesn't have "description" prop when rendered in the horizontal manner but while inside the "More" menu it's whole layout changes and it also needs to render that "description", where will we get this "description" from tho, we get it form the `items` array. 

Check the [alternative APIs section](https://github.com/razorpay/blade/blob/anu/tabnav-newapi/packages/blade/src/components/TopNav/_decisions/decisions.md#alternatives-considered-for-new-tabnav-api) for more detailed problem statement and approaches we tried out.

Reply:
looks nice and clean sir 🫡 
I've dropped a few more comments regarding this, I'm not able to currently see a case where we need to provide users the flexibility of what needs to be rendered